### DISCUSSION TITLE
Idea for fastify-userland org

### DISCUSSION THREAD

**Discussion by BlackHole1** [fastify/fastify#4099] [Ideas]
Hey folks. Do you have an opinion about the creation of `fastify-userland` org?

I've found the `fastify` community to be quite active, with many people developing a variety of plugins, but the rate of subsequent iterations is very low, and the response time to issues is not very fast.
It would be great if we could create `fastify-userland` org to manage the community's plugins or other projects around `fastify`.
See what electron has done: https://github.com/electron-userland/welcome

For example, a project like this one: https://github.com/BlackHole1/fastify-workflows-community. could be placed under the `fastify-userland` organization

The benefits of this organization:
1. Improving the quality of the community project code
2. Increase the activity of your community
3. attract more developers to join `fastify`
4. facilitate the management of some of the community's projects
5. decoupled from the `fastify` community to a certain extent, not taking up too much time of internal `fastify` members

The organization should not be run entirely by community developers for several reasons:
1. Promotion will be much less effective
2. if the organization does something that is not good for `fastify`, it may make others think that it is fastify's point of view. Because the control is not in the `fastify` organization.

**This organization is more like a subdivision of `fastify`.**

> I have currently created this organization, see: https://github.com/fastify-userland.
If this proposal is approved, I will add some existing members of fastify to this organization and give them the highest administrative privileges.

**Comment by Eomm** [comment]
Hi,
I like the idea. 
In the past, there were some community plugins that were largely used but not maintained. In these cases, the projects have been transferred or forked by the fastify org.

We did already something similar, where a new member has been added to a dedicated GitHub team, but this action adds more tasks to do to deal with permissions and if the maintainer fades away, the issue triage is up to the fastify-plugin team. This is the main concern about migrating a bunch of projects in this org.

Another use case is the `fastify-workflows-community` you mentioned: the fastify team can't implement and support everything, but someone else could be an expert in that field and is not interested in joining the fastify org through its process.

In my opinion, this `fastify-userland` org could:

- develop tooling that the fastify org can't maintain
- incubating those packages that are not maintained and are not suitable for the fastify org
- teach devs to join the fastify community (PR reviews etc..)

> I will add some existing members of fastify to this organization 

I'm not sure about it, because the target for this new org would be to remove a burden in the actual team. Personally, I would not be able to help right now.

Finally, I think this new community must adhere to the fastify org's CODE_OF_CONDUCT (adapted of course) and implement its own GOVERNANCE model.
Starting a community is not easy at all, for sure a call to action is good to go but, as in politics, a to-do list would help to define some targets:

- are there any unmaintained fastify plugins?
- what are the tools that the community need and are not provided by the fastify org? (`pnpm`, `yarn`, others?)

**Reply by BlackHole1** [reply]
> I'm not sure about it, because the target for this new org would be to remove a burden in the actual team. Personally, I would not be able to help right now.

I point this out because fastify-userland, as a new organization, is decoupled to some extent from fastify. But it still needs to rely on fastify. As I said before it is more of a sub-division.

Joining this new organization does not make your job any more onerous as your main role is to help fastify-userland to give a degree of help (advice) on things that are uncertain and to prevent fastify-userland from going down the wrong path.

Just like: https://github.com/electron/governance/blob/main/wg-administrative/README.md

**Reply by BlackHole1** [reply]
> are there any unmaintained fastify plugins?

see: https://www.fastify.io/ecosystem/  - `Community Plugins`. Click on a few at random and you will see that some of them are no longer maintained.

> what are the tools that the community need and are not provided by the fastify org? (pnpm, yarn, others?)

1. fastify existing projects that don't want to be maintained but are used by users.
2. individual developers who find it tiring to maintain and no longer want to maintain the project alone
3. projects that are no longer maintained by individual developers because they have changed companies and no longer use fastify
4. projects that are not suitable for fastify maintenance but are needed by the community

**Comment by Uzlopak** [comment]
@jsumners

**Reply by jsumners** [reply]
There was some discussion about this idea elsewhere and I do believe I protested on the grounds that it would create confusion and require extra time from the already over burdened core maintainers.

**Reply by Uzlopak** [reply]
I was just searching regarding fastify-userland and wanted to add you. But I get your point. 

Ferengi Rule of acquisition Nr. 239: Never be afraid to mislabel a product.

**Reply by BlackHole1** [reply]
I can well understand what you are worried about.
Maybe I can rename the org to something like: `@fastify-community`  and explain this in the org github profile?

Do you guys have any other suggestions? Because this problem has never been solved. So far I haven't created a profile in `@fastify-userland`.

**Reply by jsumners** [reply]
In my opinion, any prefix of `@fastify<something>` will result in confusion among users about who maintains the modules.

**Reply by BlackHole1** [reply]
OK. I'll rename it to: `nodejs-web-server-userland`  tomorrow, and if anyone has any ideas in the meantime, we can discuss them again.

**Reply by Eomm** [reply]
`communitify`?

**Reply by jsumners** [reply]
I suppose what I am saying is this: there's no way to create a community organization that:

1. Is recognizable as a Fastify community
2. Does not imply the the core Fastify team is involved
